nexusutils/dataconverter/readers/em_spctrscpy/utils/em_example_data.py

```diff
@@ -0,0 +1,416 @@
+#!/usr/bin/env python3
+"""Utility functions for generation of EM NeXus datasets for dev purposes."""
+
+# -*- coding: utf-8 -*-
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=E1101, R0801
+
+import hashlib
+
+import datetime
+
+import math
+
+import numpy as np
+
+# import matplotlib.pyplot as plt
+
+import hyperspy.api as hs
+
+from ase.data import chemical_symbols
+
+from nexusutils.dataconverter.readers.em_spctrscpy.utils.em_versioning \
+    import NX_EM_ADEF_NAME, NX_EM_ADEF_VERSION, NX_EM_EXEC_NAME, NX_EM_EXEC_VERSION
+
+# parameter affecting reconstructed positions and size
+# ##MK default parameter
+MAX_USERS = 4
+# compose hypothetical spectrum (ignoring background)
+XRAY_ENERGY_DELTA = 0.01  # keV
+PEAK_WIDTH_SIGMA = 0.1  # stddev assumed the same for all peaks (keV)
+SIGNAL_YIELD = 1000.  # arbitrarily assumed signal intensity
+
+
+class EmSpctrscpyCreateExampleData:
+    """A synthesized dataset meant to be used for development purposes only!."""
+
+    def __init__(self, n_entries):
+        # assure deterministic behaviour of the PRNG
+        np.random.seed(seed=10)
+
+        self.n_entries = 1
+        if n_entries >= 1:
+            if n_entries <= 1000:
+                self.n_entries = n_entries
+        print("Generating " + str(self.n_entries) + " random example NXem entries...")
+        self.entry_id = 0
+        self.elements_observed = []
+        self.e_axis = []
+        self.cnts_summary = []
+        # synthesizing realistic datasets for electron microscopy requires a
+        # physical model of the image formation process see e.g. Kirkland et al.
+        # for an (incomplete) view of this large topic
+        # https://doi.org/10.1007/978-3-030-33260-0
+        # we are interested in having spectra and maybe some synthesized images
+        # as this is already sufficient to explore search capabilities in e.g. OASIS
```

Comment:
Consider placing such details about the methodology in the class docs. Again this will be outputted in any docs we create. This looks like important information.